datahub-web-react/src/app/lineage/__tests__/constructTree.test.ts

```diff
@@ -397,6 +407,64 @@ describe('constructTree', () => {
                     status: null,
                     platform: airflowPlatform,
                     subtype: undefined,
+                    canEditLineage: false,
+                },
+            ],
+        });
+    });
+
+    it('should construct a tree taking into account updatedLineages in state', () => {
```

Comment:
This whole updated lineages pattern is a bit tricky to grasp conceptually

Reply:
yeah agreed but tbh my hands were kind of tied with how this lineage viz renders things..